Add Devise

Wahaj-Ali opened the pull request:
## Highlights
- Use devise gem for authentication.
- Now the user can 
  -  register itself.
  -  log in with a combination of email and password.
  - reset the password.
- Hashed passwords are stored in the database.

Wahaj-Ali committed:
Update application_controller.rb

Remove current_user.

Wahaj-Ali committed:
Add Devise to Gemfile.

Wahaj-Ali committed:
Update application.html.erb

Add navbar with links to sign-in, sign-out and edit account.

Wahaj-Ali committed:
Update application_controller.rb

Add controller filters & helpers.

Wahaj-Ali committed:
Update users_controller.rb

Add before_action helper for users to sign in before viewing the posts.

Wahaj-Ali committed:
Update new.html.erb

Add a input field for the user to add a name in the sign-up form.

Wahaj-Ali committed:
Update new.html.erb

Add placeholders for input fields of name and password.

Wahaj-Ali committed:
Update schema.rb

Wahaj-Ali committed:
Update seeds.rb

codehass committed:
Add flesh to application.html

codehass committed:
Add configurations

codehass committed:
Generate devise

Wahaj-Ali committed:
Add stules to the registration form and the newly added navbar.

Wahaj-Ali committed:
Remove linter errors.

Wahaj-Ali committed:
Add styles

Wahaj-Ali committed:
Remove styleint errors.

Felix45 commented on app/models/user.rb:
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
- [ Required ] According to the project requirements when a user creates an account the application should ask the user to confirm the email address they used to sign up. Please consider adding the `confirmable` attribute of `devise` to your `user` model to enable this.

Wahaj-Ali replied:
Issue has been solved.

Felix45 commented on app/views/devise/sessions/new.html.erb:
```diff
@@ -0,0 +1,26 @@
+<h2 class="login-heading">Log in</h2>
+
+<%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email", placeholder: "username" %>
+  </div>
+
+  <div class="field">
+    <%= f.label :password %><br />
+    <%= f.password_field :password, autocomplete: "current-password", placeholder: "password" %>
+  </div>
+
+  <% if devise_mapping.rememberable? %>
+    <div class="field" id="remem">
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
```
- [ Required ] I think it would be better if the design of your `registration` and `login` forms appear exactly as they are in the project wireframes for this milestone. As a developer, you'll be required to bring out designs just as they appear in design documents. Kindly see the screenshots below and rectify this.

## Project wireframe
<img src="https://github.com/microverseinc/curriculum-rails/raw/main/blog-app/images/blog_login.png" width="375px" />

## Your design 
<img src="https://user-images.githubusercontent.com/16392046/231541577-897822f6-24a1-4800-b590-81f02dc17422.png" width="375px" />

Wahaj-Ali replied:
Modified the design ✅

Felix45 reviewed (changes requested):
Hi @Wahaj-Ali , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<img src="https://camo.githubusercontent.com/56bcd754643682fcc078b66eaa7278dead6c9c5013595c875b4ddcbe0f0ef0aa/68747470733a2f2f6d65646961302e67697068792e636f6d2f6d656469612f78554f786668365a4d373565664d3342716f2f67697068792d646f776e73697a65642d6d656469756d2e676966" width="375px" />

## Highlights :heavy_check_mark: 
- [x] Good code organization :white_check_mark: 
- [x] No linter errors :white_check_mark: 
- [x] Repository has a professional readme file :white_check_mark: 
- [x] Descriptive pull request title and summary :white_check_mark:   

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Wahaj-Ali committed:
Made required changes.

In this commit
- Modify the style to match the wireframe.
-  add the confirmable attribute of devise to the user model.

Wahaj-Ali committed:
Update application.css

Remove styleint errors.

Wahaj-Ali committed:
Update application.css

Reem-lab reviewed (approved):
# Hi @Wahaj-Ali,👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🔥🔥💯💯
<img src="https://media2.giphy.com/media/3o6fJ1BM7R2EBRDnxK/giphy.gif"/>

## Highlights 🔥🔥🔥🔥
- Followed Gitflow ✅
- Great working on the functionality 🔥🔥
- All requested changes are done✅🤩




## Optional suggestions



_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Wahaj-Ali committed:
Modify devise config environment to allow email confirmation.
